feat: add initial implementation

marstamm opened the pull request:
### Proposed Changes
This PR adds the initial implementation for the variable outline bottom panel.

![image](https://github.com/user-attachments/assets/05259614-4165-467d-8917-a62655a9bcb2)

This iteration includes:

- Search for variables, including filter for "written variables only"
- Bidirectional selection of Tasks, including multi-select
- Sortable Table, clickable origin/scope/name for easy filtering
- Shipping as a React component to be integrated by desktop modeler

You can try out the demo with `npm run start` or download the artifact with the [test integration](https://github.com/camunda/camunda-modeler/tree/integrate-variable-outline) of the camunda modeler:

- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/integrate-variable-outline/camunda-modeler-integrate-variable-outline-linux-x64.tar.gz
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/integrate-variable-outline/camunda-modeler-integrate-variable-outline-mac.dmg
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/integrate-variable-outline/camunda-modeler-integrate-variable-outline-mac.zip
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/integrate-variable-outline/camunda-modeler-integrate-variable-outline-win-ia32.zip
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/integrate-variable-outline/camunda-modeler-integrate-variable-outline-win-x64.zip

<!--

Add relevant context (issue fixed or related to), 
a capture of the UI changes (if any) as well as 
steps to try out your changes.

--> 

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [x] **Visual demo** attached
* [x] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [ ] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

<!--

Thanks for creating this pull request! ❤️

-->

barmac commented:
I can't access the mac artifacts. Can you rebuild it?

marstamm commented:
That's weird, the Job succeeded. Maybe they have really short retention on S3 🤔 

I [triggered the build again](https://github.com/camunda/camunda-modeler/actions/runs/10145337805/job/28221730331), should be ready in a few minutes

marstamm commented:
Never mind, the artifact links for mac are outdated on the build-on-demand site. They don't include the Architecture 😄  

Here are working links:

https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/integrate-variable-outline/camunda-modeler-integrate-variable-outline-mac-x64.zip
https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/integrate-variable-outline/camunda-modeler-integrate-variable-outline-mac-arm64.zip

I will file a PR tomorrow to fix them

barmac commented:
> That's weird, the Job succeeded. Maybe they have really short retention on S3 🤔
> 
> I [triggered the build again](https://github.com/camunda/camunda-modeler/actions/runs/10145337805/job/28221730331), should be ready in a few minutes

3 weeks if I recall correctly

barmac commented:
The help button tooltip causes a scrollbar to appear:

https://github.com/user-attachments/assets/aad16a5b-71f0-4de3-8f19-feeeaf936e54

barmac commented:
There's a small difference of height between the top bar elements:

https://github.com/user-attachments/assets/43829b44-cfa1-4568-b9fd-4a5726160bf9

barmac commented:
I'd expect the plugin to scroll to the nested element when entry is clicked:

https://github.com/user-attachments/assets/03db0e89-ddb3-4e89-99d0-f5880656305c

barmac commented:
The tooltip is cut-off:
![image](https://github.com/user-attachments/assets/2dd312b6-cecb-40f4-8f26-550a4ab2c346)

barmac commented:
Generally, the table on the right-hand side looks cut-off. Perhaps a border could improve it.

barmac commented on example/App.css:
```diff
@@ -0,0 +1,47 @@
+  border: 1px solid #ccc;
+}
+
+/* #root {
```
We could remove the styles we don't need.

barmac commented on example/App.jsx:
```diff
@@ -0,0 +1,36 @@
+  const [ content, setContent ] = useState(null);
+
+  return (
+    <>
```
This seems to be unnecessary given we wrap everything in the Theme.

barmac commented on example/BottomPanel/bottom-panel.css:
```diff
@@ -0,0 +1,9 @@
+.bottom-panel {
+  width: 100%;
+  background-image: url('./Bottom_panel_left.png');
```
Great solution for the example :D

marstamm replied:
:hackerman:

barmac commented:
By clicking on the Scope, I can break the modeler:

https://github.com/user-attachments/assets/df7c610f-695d-46a4-8af2-11580e222d2d

barmac commented:
Elements list doesn't display anything in a collaboration diagram - is this intended?
<img width="1415" alt="image" src="https://github.com/user-attachments/assets/cf9d2d3f-471d-4afe-b0e7-ff2d1d6b1b34">

barmac commented:
Apart from the comments above, the tool works as supposed.

marstamm commented:
Thank you for your review, great finding! I'll look into them

marstamm commented:
I addressed the review hints and updated the modeler artifacts. Ready for another look 👀

> Generally, the table on the right-hand side looks cut-off. Perhaps a border could improve it.

I added 5px border, now the table separators do not go all the way to the scroll bar

> Elements list doesn't display anything in a collaboration diagram - is this intended?

Nope, should be fixed now

barmac commented:
> I'd expect the plugin to scroll to the nested element when entry is clicked:
> 
>  Screen.Recording.2024-08-02.at.08.55.58.mov

This is still not fixed for the origin:


https://github.com/user-attachments/assets/52878588-9454-4ecc-a2b9-42a61f9d8973

barmac commented:
Apart from the issue above, the rest of my comments are resolved. Great work on this.

marstamm commented:
I missed that 🙈 


Should be fixed with [2ccc73c](https://github.com/bpmn-io/variable-outline/pull/1/commits/2ccc73c749415e5bd8768dda209fda30a1b766cc)
